katsdptelstate/telescope_state.py

```diff
@@ -35,11 +37,13 @@
 class TelescopeState:
     """Interface to attributes and sensors stored in a Redis database.
 
-    There are two types of keys permitted: single immutable values, and mutable
-    keys where the full history of values is stored with timestamps. These are
-    mapped to the Redis string and zset types. A Redis database used with this
-    class must *only* be used with this class, as it does not deal with other
-    types of values.
+    There are three types of keys permitted: single immutable values,
+    indexed immutable values, and mutable keys where the full history of values
```

Comment:
This confused me slightly - the sub-keys are immutable not the whole index - this kind of sounds like you can only set the entire hash in a one-shot way.

Reply:
I started out calling these keys "indexed immutable" but changed to "indexed" later - this was a leftover.

I am however strongly tempted to rip out this whole paragraph and just refer readers to the README instead.

Reply:
In fact, I'm going to do just that.